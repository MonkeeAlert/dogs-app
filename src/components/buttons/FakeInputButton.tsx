import {Pressable, StyleSheet, Text, View} from 'react-native';
import React, {ReactElement} from 'react';

interface Props {
  children: ReactElement;
  onPress: () => void;
}

const FakeInputButton = ({children, onPress}: Props) => {
  const handleRedirect = () => {
    if (onPress || typeof onPress === 'function') {
      return onPress();
    }
  };

  return (
    <Pressable testID={'FakeInputButton_Pressable'} onPress={handleRedirect}>
      <View>{children}</View>
    </Pressable>
  );
};

export default FakeInputButton;

const styles = StyleSheet.create({});
