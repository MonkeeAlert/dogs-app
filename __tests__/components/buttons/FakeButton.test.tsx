import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import FakeInputButton from '../../../src/components/buttons/FakeInputButton';
import {render, fireEvent, act} from '@testing-library/react-native';

describe('FakeButton', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <FakeInputButton action={() => console.log('@foo')}>
          <Text>foo</Text>
        </FakeInputButton>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('does nothing when action was not provided', () => {
    const {getByTestId} = render(
      <FakeInputButton>
        <Text>foo</Text>
      </FakeInputButton>,
    );
    const view = getByTestId('FakeInputButton_Pressable');

    act(() => {
      const evt = fireEvent(view, 'onPress');

      expect(evt).toBe('No action provided');
    });
  });

  it('does something when action was provided', () => {
    const mockedFn = jest.fn().mockImplementation(() => true);

    const {getByTestId} = render(
      <FakeInputButton action={mockedFn}>
        <Text>foo</Text>
      </FakeInputButton>,
    );
    const view = getByTestId('FakeInputButton_Pressable');

    act(() => {
      const evt = fireEvent(view, 'onPress');

      expect(evt).toBe(true);
    });
  });
});
