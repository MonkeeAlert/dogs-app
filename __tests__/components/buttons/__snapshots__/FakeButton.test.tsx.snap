// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeButton matches snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="FakeInputButton_Pressable"
>
  <View>
    <Text>
      foo
    </Text>
  </View>
</View>
`;
